src/ResourceManager/Sql/Commands.Sql/Instance Failover Group/Cmdlet/SetAzureSqlInstanceFailoverGroup.cs

```diff
@@ -0,0 +1,206 @@
+        [ValidateNotNullOrEmpty]
+        [ValidateRange(1, int.MaxValue/60)]
+        [PSDefaultValue(Help = "1")]
+        public int GracePeriodWithDataLossHours { get; set; }
+
+        /// <summary>
+        /// Gets or sets the failover policy for read only endpoint of the Sql Azure Instance Failover Group.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Whether outages on the secondary server should trigger automatic failover of the read-only endpoint. This feature is not yet supported.")]
+        [ValidateNotNullOrEmpty]
+        public AllowReadOnlyFailoverToPrimary AllowReadOnlyFailoverToPrimary { get; set; }
+
+        /// <summary>
+        /// Get the entities from the service
+        /// </summary>
+        /// <returns>The list of entities</returns>
+        protected override IEnumerable<AzureSqlInstanceFailoverGroupModel> GetEntity()
+        {
+            if (InputObject != null)
+            {
+                Location = InputObject.Location;
+                Name = InputObject.Name;
+                ResourceGroupName = InputObject.ResourceGroupName;
+            }
+            else if (!string.IsNullOrWhiteSpace(ResourceId))
+            {
+                ResourceIdentifier identifier = new ResourceIdentifier(ResourceId);
+                Location = identifier.ResourceName;
+                identifier = new ResourceIdentifier(identifier.ParentResource);
+                Name = identifier.ResourceName;
+                identifier = new ResourceIdentifier(identifier.ParentResource);
+                ResourceGroupName = identifier.ResourceName;
+            }
+
+            return new List<AzureSqlInstanceFailoverGroupModel>() {
+                ModelAdapter.GetInstanceFailoverGroup(this.ResourceGroupName, this.Location, this.Name)
+            };
+        }
+
+        /// <summary>
+        /// Create the model from user input
+        /// </summary>
+        /// <param name="model">Model retrieved from service</param>
+        /// <returns>The model that was passed in</returns>
+        protected override IEnumerable<AzureSqlInstanceFailoverGroupModel> ApplyUserInputToModel(IEnumerable<AzureSqlInstanceFailoverGroupModel> model)
+        {
+            List<AzureSqlInstanceFailoverGroupModel> newEntity = new List<AzureSqlInstanceFailoverGroupModel>();
+            AzureSqlInstanceFailoverGroupModel newModel = model.First();
+            object parameterValue;
+            FailoverPolicy effectivePolicy = FailoverPolicy;
+            if (!MyInvocation.BoundParameters.ContainsKey("FailoverPolicy"))
+            {
+                // If none was provided, use the existing policy.
+                Enum.TryParse(newModel.ReadWriteFailoverPolicy, out effectivePolicy);
+            }
+
+            int? gracePeriod = null;
+            if (!FailoverPolicy.ToString().Equals("Manual"))
```

Comment:
@jugeorge same comment here about checking to see if `FailoverPolicy` is provided before accessing properties